File: 프로그래머스/소수찾기.cpp
#include <string>
#include <vector>
#include <math.h>
using namespace std;

bool isPrime(int n)
{
	static vector<int> primes{ 2,3,5,7 };
	int rootNumber;
	int i, j = 0;
	if (n == 1)
		return false;
	if (n == 2) {
		primes.push_back(n);
		return true;
	}
	if (n % 2 == 0)
		return false;
	rootNumber = sqrt(n);
	while (i <= rootNumber) {
		i = primes[j++];
		if (n%i == 0)
			return false;

	}
	for (; i <= rootNumber; i += 2)
	{
		if (n%i == 0)
			return false;
	}
	primes.push_back(n);
	return true;

}

int solution(int n) {
	int answer = 0;
	vector<bool> seive(n + 1);
	for (int i = 2; i <= n; i++) {
		if (seive[i] == false)
		{
			answer++;
		}
		for (int j = i; j <= n; j += i)
		{
			seive[j] = true;
		}
	}
	return answer;

}

File: SORT/소수찾기.py
import sys
import math
sys.stdin = open("input.txt","r")
input = sys.stdin.readline

def isPrime(num):
    if num == 2 or num == 3:
        return True
    for i in range(2,int(math.sqrt(num))):
        if num % i == 0:
            return False
    return True


def setPrime():
    global numbers
    primes[0] = True
    primes[1] = True
    for i in range(2,1001):
        if primes[i] == False:
            if isPrime(i) == True:
                for j in range(i+i,1001,i):
                    primes[j] = True


numbers = [False] * 1001
N = int(input())

nums = list(map(int, input().split()))

setPrime()

cnt = 0
for x in nums:
    if not numbers[x]:
        cnt += 1

print(cnt)

